src/laser_geometry.cpp

```diff
@@ -437,7 +437,8 @@ const boost::numeric::ublas::matrix<double>& LaserProjection::getUnitVectors_(do
     {
       //check to see if we want to keep the point
       const float range = scan_in.ranges[i];
-      if ((preservative && std::isfinite(ranges(i,0))) || (range < range_cutoff && range >= scan_in.range_min))
```

Comment:
@v-mehta We want to draw one to one correspondences between LIDAR and point cloud data after  the processing. By removing NaNs we are kind of violating definition of "preservative".

@afakihcpr can a LIDAR have -INF too besides INF and NaN? I have not seen -Inf